### PR TITLE
chore: clean up space distribution feature in Anvil

### PR DESCRIPTION
**Pull request by marks0351**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
## Description
This PR addresses two particular things with space distribution
- Issue: Space distribution distributing more than max column limit space
- Enhancement: Replace flew grow css for zones to flex basis to make sure column widths are even irresepective of number of zones in a section.

Additional Fixes:
- replaced test id of widgets editor to abide by repo standards.
- disable crud app generation feature for Anvil.
#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enabled new feature flags to enhance application capabilities.
	- Introduced dynamic space redistribution for improved layout management.
	- Added "Generate App" functionality for users with specific feature access.
- **Enhancements**
	- Updated UI selectors for better consistency and reliability.
	- Refined layout system logic to always utilize the Anvil layout system.
- **Refactor**
	- Optimized feature flag selection logic for efficiency.
	- Improved flex property handling in layout components for more accurate space distribution.

### CODE REVIEW

**Review by riodeuno — APPROVED** [review]

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/propertyPane/PropertyPaneSectionSpaceDistributor.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
```
[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/sectionSpaceDistributor/propertyPane/PropertyPaneSectionSpaceDistributor.tsx:1]: There is a separate control for distribution, this is a component used to build the said control.

**Review by riodeuno — APPROVED** [review]

**Pull request merged**